### ISSUE TITLE
Volt crashes when navigating to some folders

### ISSUE THREAD

**Issue by JayTheCoderX** [TheRedXD/Volt#3]
**Describe the bug**
Volt crashes when navigating to some folders

**To Reproduce**
Steps to reproduce the behaviour:
```
> be me
> have a cat
> download all my downloads
> open volt
> navigate to downloads
> cry
```

**Expected behaviour**
Volt doesn't crash

**Screenshots or videos**
Screenshot of volt after it crashed:

Error log:
```
Panic occurred in file 'src/main.rs' at line 289!
Panic message: byte index 30 is not a char boundary; it is inside '\u{a0}' (bytes 29..31) of `Run scripts on every page  _  Extensions  _  Chrome for Developers_files`
error: process didn't exit successfully: `debug\volt.exe` (exit code: 101)
```

**Device Info (Run Volt with the `--info` flag and paste in the output. If it doesn't work, use any other system info tool to get this information):**
OS: windows
Desktop Environment: None
Compositor: None
CPU: Intel(R) Core(TM) i5-9400 CPU @ 2.90GHz
GPU: NVIDIA GeForce RTX 2070
OS Family: windows
OS Distribution: None
Architecture: x86_64
Version: 0.1.0

**Additional context**

**Comment by TheRedXD** [comment]
Confirmed to occur on linux as well. Will look into it.

![image](https://github.com/user-attachments/assets/2d6a4a09-d955-48bf-b252-c8be5a74c87e)

Seems to occur with unicode.

**Comment by TheRedXD** [comment]
Fixed in commit 1e3eee2, now using unicode-truncate for truncating file names
